### PR TITLE
8279997: check_for_dynamic_dump should not exit vm

### PR DESCRIPTION
**Pull request by yminqi**
Hi, please review
  When run with -Xshare:auto and the base archive could not be loaded, vm should not exit in DynamicArchive::check_for_dynamic_dump, instead, it should continue without sharing.
  The fix is the run will continue but print out warning to remind user of using -Xlog:cds for more information.

Tests: tier1,tier4

Thanks
Yumin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8279997](https://bugs.openjdk.java.net/browse/JDK-8279997): check_for_dynamic_dump should not exit vm

### Reviewers
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**)
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7433/head:pull/7433` \
`$ git checkout pull/7433`

Update a local copy of the PR: \
`$ git checkout pull/7433` \
`$ git pull https://git.openjdk.java.net/jdk pull/7433/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7433`

View PR using the GUI difftool: \
`$ git pr show -t 7433`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7433.diff">https://git.openjdk.java.net/jdk/pull/7433.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/ArchiveConsistency.java [hunk 1]
```diff
@@ -202,5 +202,30 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
...
+        run(topArchiveName,
+            "-Xshare:auto",
+            "-XX:+AutoCreateSharedArchive",
+            "-cp",
+            appJar, mainClass)
+            .assertNormalExit(output -> {
+                output.shouldContain("warning: -XX:+AutoCreateSharedArchive is unsupported when base CDS archive is not loaded");
+            });
+
+        startTest("10. -XX:SharedArchiveFile=" + topArchiveName + " -XX:ArchiveClassesAtExit=" + getNewArchiveName("top3"));
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/ArchiveConsistency.java:221]: Should this be `startTest("11. ...` ?

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java [hunk 2]
```diff
@@ -198,32 +187,36 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
 
         testcase("Specifying -XX:+RecordDynamicDumpInfo should not cause dynamic dump");
         run2(baseArchiveName, null,
-             "-XX:+RecordDynamicDumpInfo",
-             "-Xlog:cds+dynamic=debug",
-             "-cp", appJar, mainClass)
+            "-XX:+RecordDynamicDumpInfo",
+            "-Xlog:cds+dynamic=debug",
+            "-cp", appJar, mainClass)
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java:192]: This change seems unnecessary.

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java [hunk 3]
```diff
@@ -198,32 +187,36 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
...
-                });
+            String ERROR = "-XX:ArchiveClassesAtExit is unsupported when base CDS archive is not loaded";
+            dump2(baseArchiveName,
+                  topArchiveName,
+                  "-Xshare:off",
+                  "-Xlog:cds",
+                  "-cp", appJar, mainClass)
+            .assertNormalExit(output -> {
+                output.shouldNotMatch("\\[cds,dynamic");
+                output.shouldContain("-XX:ArchiveClassesAtExit is unsupported when base CDS archive is not loaded");
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java:207]: You could use the `ERROR` defined at line 199 above.

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java [hunk 4]
```diff
@@ -198,32 +187,36 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
...
-                          "-cp", appJar, mainClass);
-                });
+            dump2(baseArchiveName + ".notExist",
+                  topArchiveName,
+                  "-Xlog:cds",
+                  "-Xshare:auto",
+                 "-cp", appJar, mainClass)
+            .assertNormalExit(output -> {
+                output.shouldNotMatch("\\[cds,dynamic");
+                output.shouldContain("-XX:ArchiveClassesAtExit is unsupported when base CDS archive is not loaded");
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java:218]: You could use the `ERROR` defined at line 199 above.

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java [hunk 5]
```diff
@@ -233,7 +226,7 @@ private static void doTest(String baseArchiveName, String topArchiveName) throws
                   "-Xshare:auto",
                   "-Xlog:cds",
                   "-cp", appJar, mainClass)
-                .assertAbnormalExit("Cannot use the following option when dumping the shared archive: --patch-module");
+            .assertAbnormalExit("Cannot use the following option when dumping the shared archive: --patch-module");
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/SharedArchiveFileOption.java:229]: This change is unnecessary.

**Review by calvinccheung — COMMENTED** [review]
Looks good. Some nits on the tests.

**Review by iklam — APPROVED** [review]
LGTM.

**Review by calvinccheung — APPROVED** [review]
Updated webrev looks good.

**Comment by yminqi** [comment]
@calvinccheung @iklam Thanks for review!

**Pull request closed without merging**